onadata/apps/api/tasks.py

```diff
@@ -145,3 +150,34 @@ def send_project_invitation_email_async(
     else:
         email = ProjectInvitationEmail(invitation, url)
         email.send()
+
+
+@app.task(track_started=True)
+def regenerate_form_instance_json(xform_id: int):
+    """Regenerate a form's instances json
+
+    Json data recreated afresh and any existing json data is overriden
+    """
+    try:
```

Comment:
Can we wrap this block in a condition such that if `xform.is_instance_json_regenerated` is True then we dont need to execute it